src/ber/de.rs

```diff
@@ -454,6 +454,17 @@ impl<'input> crate::Decoder for Decoder<'input> {
             Ok(buffer)
         }
     }
+    fn decode_fixed_octet_string<const N: usize>(
+        &mut self,
+        tag: Tag,
+        constraints: Constraints,
+    ) -> Result<[u8; N], Self::Error> {
+        // BER is not aware of the constraints
+        let data = self.decode_octet_string(tag, constraints)?;
+        let mut array = [0u8; N];
+        array.copy_from_slice(data.as_slice());
+        Ok(array)
```

Comment:
BER is not constraint aware, meaning that it doesn't make any format optimisations, but we can still optimise and check this case. If we're parsing a fixed length octet we can still both check that the length matches the fixed length constraint, and copy directly from the input into the array, skipping the vec here.

Reply:
Good point!